### ISSUE TITLE
@ts-check complains about a require of a JSON file.

### ISSUE THREAD

**Issue by egamma** [Microsoft/TypeScript#21262]
The node `require` can also read in  .json files. 

In the following code the require is flagged with an error when the ´test.json´ file exists, but the code executes correctly.

```javascript
//@ts-check
let test = require("./test.json")
```

![image](https://user-images.githubusercontent.com/172399/35092256-1762afac-fc3f-11e7-9280-df81f82f15b5.png)

**Comment by mhegazy** [comment]
Today the compiler does not recognize json files, neither in terms of resolution/lookup nor in terms of content. The suggestion is tracked by https://github.com/Microsoft/TypeScript/issues/7071

We have a fix for that in https://github.com/Microsoft/TypeScript/pull/13665, it is a bit stale now, so I do not have a concrete ETA at the moment.
